packages/docusaurus-search-local/src/server/parse.ts

```diff
@@ -124,8 +124,9 @@ export function html2text(
           // <a class="hash-link" href="#first-header" title="Direct link to heading">#</a>
           .not("a[aria-hidden=true], a.hash-link")
           .text();
-        const hash = $(heading).find("a.hash-link").attr("href") || "";
-
+        const linkHash = $(heading).find("a.hash-link").attr("href") || "";
+        const [, hashPart] = linkHash.split("#");
+        const hash = hashPart ? `#${hashPart}` : "";
```

Comment:
Looks better than the code before assuming an anchor never includes a pathname

Please note that an anchor can contain `#` characters 😄 

![CleanShot 2024-08-13 at 13 33 24](https://github.com/user-attachments/assets/1d81564f-3a0c-4df2-b3be-1a3d385fd3d9)

Reply:
Good point, @slorber , thx for pointing it out! Didn't think about that. Do you think using new URL().hash could fix it?